app/src/androidTest/java/com/github/h3lp3rs/h3lp/forum/ForumAnswersActivityTest.kt

```diff
@@ -0,0 +1,83 @@
+package com.github.h3lp3rs.h3lp.forum
+
+import android.content.Intent
+import androidx.test.core.app.ActivityScenario
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.action.ViewActions.click
+import androidx.test.espresso.action.ViewActions.replaceText
+import androidx.test.espresso.intent.Intents.init
+import androidx.test.espresso.intent.Intents.release
+import androidx.test.espresso.matcher.ViewMatchers.withId
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.github.h3lp3rs.h3lp.R
+import com.github.h3lp3rs.h3lp.USER_TEST_ID
+import com.github.h3lp3rs.h3lp.forum.data.ForumPostData
+import com.github.h3lp3rs.h3lp.signin.SignInActivity
+import junit.framework.Assert.assertEquals
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mockito
+import org.mockito.kotlin.any
+import java.time.ZonedDateTime
+
+const val CATEGORY_TEST_STRING = "GENERAL"
+val CATEGORY_TEST = ForumCategory.GENERAL
+const val QUESTION_TEST = "question"
+const val ANSWER_TEST = "answer"
+
+@RunWith(AndroidJUnit4::class)
+class ForumAnswersActivityTest {
+    private val forumPosts: MutableMap<String, List<String>> = mutableMapOf()
+
+    @Before
+    fun setup() {
+        val intent = Intent(
+            ApplicationProvider.getApplicationContext(),
+            ForumAnswersActivity::class.java
+        ).apply {
+            putExtra(EXTRA_FORUM_CATEGORY, CATEGORY_TEST_STRING)
+        }
+
+        SignInActivity.setName(USER_TEST_ID)
+
+        val forum = Mockito.mock(Forum::class.java)
+        ForumCategory.setForum(CATEGORY_TEST,forum)
+        Mockito.`when`(forum.newPost(any(), any())).then {
+            val content = it.getArgument<String>(1)
+            forumPosts[QUESTION_TEST] = listOf(content)
+            return@then any()
+        }
+        Mockito.`when`(forum.child(any() as String)).thenReturn(forum)
```

Comment:
Static import, plus you can define an alias for When instead of writing weird 'when'